backend/uclapi/oauth/scoping.py

```diff
@@ -0,0 +1,69 @@
+# Storage of the scope map
+# The purpose of this setup is that the OAuth scope of any app can be stored in a single field,
+# and it can be added to in the future.
+# We have a BigIntegerField to work with, which means 64 bits of storage. This translates into
+# 64 types of scope, each of which can be checked with a bit mask.
+# E.g. roombookings has scope 0, which is 0000000000000000000000000000000000000000000000000000000000000001b.
+# This is because the 0th bit (LSB) is set to 1.
+# Room bookings + UCLU = 1001b, or a scope number of 11
+
+class Scopes:
+    SCOPE_MAP = {
+        "roombookings": (0, "Private room bookings data"),
+        "timetable": (1, "Private timetable data"),
+        "uclu": (2, "Private UCLU data"),
+        "moodle": (3, "Private Moodle data")
+    }
+
+    # Add a scope to the scope number given and return the new number
+    def add_scope(self, current, scope_name):
+        try:
+            scope_shift = self.SCOPE_MAP[scope_name][0]
+        except KeyError:
+            return current
+
+        return (current | (1 << scope_shift))
+
+    # Check whether a scope is present in the current scope number given
+    def check_scope(self, current, scope_name):
+        try:
+            scope_shift = self.SCOPE_MAP[scope_name][0]
+        except KeyError:
+            return False
+
+        return ((1 << scope_shift) & current) > 0
+
+    # Remove a scope from the current scope number
+    def remove_scope(self, current, scope_name):
+        try:
+            scope_shift = self.SCOPE_MAP[scope_name][0]
+        except KeyError:
+            return current
+

```

Comment:
The lines below might be better written as `current & ~(1 << scope_shift)`. If you try to unset a bit which is NOT set, nothing will happen anyways.